### DISCUSSION TITLE
Foxglove-native JointState schema equivalent

### DISCUSSION THREAD

**Discussion by amacneil** [foxglove/community#439] [Q&A]
Foxglove alternative to ROS' [sensor_msgs/JointState](http://docs.ros.org/en/noetic/api/sensor_msgs/html/msg/JointState.html).

**Comment by defunctzombie** [comment]
Do we want to capture other functionality beyond what the ROS message provides for? Might be worth having some users who don't use ROS but have their own similar messages sharing what those look like so we can learn about the alternative ideas.

**Comment by johannespitz** [comment]
FYI: We have two readings of the joint angles (position encoder and increment counter), temperatures, motor pwm, and multiple error indicators for each joint.

**Reply by banisadr** [reply]
Do you capture all of this state in a custom schema of our own today?

**Reply by johannespitz** [reply]
Yes, right now we store everything, and sometimes generate plots in python (without using any foxglove tools).
For real time supervision of the system in foxglove studio it would be awesome to link the fields to joints, and visualize, e.g. error indicators directly in the 3d view of the urdf. Measurements could show up on click or while hovering over the joint.

**Reply by johannespitz** [reply]
Ideally, the joint angles could also directly be used to animate the 3d view. (Without the need to compute transforms by hand.)
https://github.com/orgs/foxglove/discussions/454
